Duplicati/Library/Main/Controller.cs

```diff
@@ -671,21 +671,18 @@ private void SetupCommonOptions(ISetCommonOptions result, ref string[] paths, re
             foreach (Library.Interface.IGenericModule m in DynamicLoader.GenericLoader.Modules)
                 m_options.LoadedModules.Add(new KeyValuePair<bool, Library.Interface.IGenericModule>(Array.IndexOf<string>(m_options.DisableModules, m.Key.ToLower()) < 0 && (m.LoadAsDefault || Array.IndexOf<string>(m_options.EnableModules, m.Key.ToLower()) >= 0), m));
 
-            var conopts = new Dictionary<string, string>(m_options.RawOptions);
+            // Make the filter read-n-write able in the generic modules
+            var pristinefilter = string.Join(System.IO.Path.PathSeparator.ToString(), FilterExpression.Serialize(filter));
+            m_options.RawOptions["filter"] = pristinefilter;
+            
             var qp = new Library.Utility.Uri(m_backend).QueryParameters;
             foreach(var k in qp.Keys)
-                conopts[(string)k] = qp[(string)k];
-
-            // Make the filter read-n-write able in the generic modules
-            var pristinefilter = conopts["filter"] = string.Join(System.IO.Path.PathSeparator.ToString(), FilterExpression.Serialize(filter));
+                m_options.RawOptions[(string)k] = qp[(string)k];

```

Comment:
I wanted to put all the options into one dictionary so that the rest of the configuration code could treat them the same.

I wasn't sure whether URL options should override command line parameters or vice-versa. I'm not sure what the use case would be to supply an option in both the URL and as a command line parameter. I can switch it around so that the URL options override command line parameters rather?